test/CadlRanchProjects/type/property/additional-properties/src/Generated/Models/ExtendsFloatAdditionalProperties.Serialization.cs

```diff
@@ -67,7 +67,10 @@ internal static ExtendsFloatAdditionalProperties DeserializeExtendsFloatAddition
                     id = property.Value.GetSingle();
                     continue;
                 }
-                additionalPropertiesDictionary.Add(property.Name, property.Value.GetSingle());
+                if (property.Value.ValueKind == JsonValueKind.Number)
+                {
+                    additionalPropertiesDictionary.Add(property.Name, property.Value.GetSingle());
+                }
```

Comment:
The point of this change is that the methods on JsonElement (for example here is `GetSingle()`) will throw `InvalidOperationException` when its value kind is not a specific value, for example here, ValueKind must be `JsonValueKind.Number` otherwise we have exceptions.